src/Microsoft.AspNetCore.Http.Connections.Client/HttpConnection.cs

```diff
@@ -385,6 +391,9 @@ private async Task SelectAndStartTransport(TransferFormat transferFormat)
                     catch (Exception ex)
                     {
                         Log.TransportFailed(_logger, transportType, ex);
+
+                        transportExceptions.Add(new Exception($"{transportType} failed: {ex.Message}"));
```

Comment:
Why not just this?

```suggestion
                        transportExceptions.Add(ex);
```

Reply:
I see, you want to tag it with the transport type. I'd like to preserve the original exception in `InnerException` at least. And I don't think we should throw `Exception`.

Reply:
Perhaps a `TransportFailedException` with a `TransportName` property and wrap all these exceptions in that, tagging the transport name (and wrapping the message)?